google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -234,23 +243,68 @@ def f(*args, **kwargs):
                 + SERVING_COMMAND_STRING_CODE_APIS
             )
 
+            # Account for user-designated dependencies when
+            # setting up remote prediction
             if GITHUB_DEPENDENCY not in obj.dependencies:
                 obj.dependencies.append(GITHUB_DEPENDENCY)
 
+            dependency_installs = []
+            for dependency in obj.dependencies:
+                if dependency != GITHUB_DEPENDENCY:
+                    dependency_name = re.split(">|=|<", dependency)[0]
+                    dependency_name = f"'{dependency_name}'"
+                    dependency_installs.append(dependency_name)
+
+            dependency_installs = " ".join(dependency_installs)
+
+            serving_command_string_cli = (
+                SERVING_COMMAND_STRING_CLI_FIRST_HALF
+                + [
+                    SERVING_COMMAND_STRING_CLI_PIP_CALL
+                    + dependency_installs
+                    + SERVING_COMMAND_STRING_CLI_GITHUB_INSTALL
+                ]
+                + SERVING_COMMAND_STRING_CLI_SECOND_HALF
+            )
+
+            # Container specification
+            location_prefix = aiplatform.initializer.global_config.location.split("-")[
+                0
+            ]
+            container_location = (
+                location_prefix if location_prefix in ("us", "europe", "asia") else "us"
+            )
+
```

Comment:
Please open a ticket and add a TODO here for matching dependency version to container version.